Week3/homework/step2-2.js

```diff
@@ -2,19 +2,27 @@
 
 function threeFive(startIndex, stopIndex, threeCallback, fiveCallback) {
   const numbers = [];
-
-  // Replace this comment and the next line with your code
-  console.log(startIndex, stopIndex, threeCallback, fiveCallback, numbers);
+  for (let i = startIndex; i <= stopIndex; i++) {
+    numbers.push(i);
+  }
+  numbers.forEach(num => {
+    if (num % 15 === 0) {
+      threeCallback(num);
+      fiveCallback(num);
+    } else if (num % 3 === 0) {
+      threeCallback(num);
+    } else if (num % 5 === 0) {
+      fiveCallback(num);
+    }
```

Comment:
I think you need not write separate for loops here for creating the numbers array and then iterating it for callback logic. You can combine both these actions in a single for loop like
```
for (let i = startIndex; i <= stopIndex; i++){
  numbers.push(i);
  if(i % 5 === 0 & i % 3 === 0) {
    // and so on
  }
}
```
Having two loops here doesn't impact much in this case since the iteration is very small. However, imagine a situation where you have to loop over 1 or 10 million records. It would be better to run the loop only once, wouldnt' it?